optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -151,6 +152,23 @@
     // join(..., planningVariableContainsCached(Talk::getPeriod, (Period a, Period b) -> a.overlaps(b)))
     // get the period value range, does a cartesian product on it, so it maps every period to an overlapping periodList
     // then keep an index from every period to all talks in an overlapping period (possible the same period)
```

Comment:
Let's make sure we have proper Javadocs for these factory methods. This is public API after all.